src/main/java/com/faforever/client/replay/ReplayCardController.java

```diff
@@ -135,22 +145,31 @@ protected void onInitialize() {
             .flatMap(reviewsSummary -> reviewsSummary.scoreProperty().divide(reviewsSummary.numReviewsProperty()))
             .when(showing));
 
-    entity.flatMap(ReplayBean::teamsProperty)
+    entity.flatMap(ReplayBean::teamPlayerStatsProperty)
         .when(showing)
-        .addListener((SimpleChangeListener<Map<String, List<String>>>) this::onTeamsChanged);
+        .addListener((SimpleChangeListener<Map<String, List<GamePlayerStatsBean>>>) this::onTeamsChanged);
   }
 
-  private void onTeamsChanged(Map<String, List<String>> teams) {
+  private void onTeamsChanged(Map<String, List<GamePlayerStatsBean>> teams) {
     teamsContainer.getChildren().clear();
-    teams.forEach((id, team) -> {
+    teams.forEach((team, playerStats) -> {
       VBox teamBox = new VBox();
 
-      String teamLabelText = id.equals("1") ? i18n.get("replay.noTeam") : i18n.get("replay.team", Integer.parseInt(id) - 1);
+      String teamLabelText = team.equals("1") ? i18n.get("replay.noTeam") : i18n.get("replay.team", Integer.parseInt(team) - 1);
       Label teamLabel = new Label(teamLabelText);
       teamLabel.getStyleClass().add("replay-card-team-label");
       teamLabel.setPadding(new Insets(0, 0, 5, 0));
       teamBox.getChildren().add(teamLabel);
-      team.forEach(player -> teamBox.getChildren().add(new Label(player)));
+
+      playerStats.forEach(player -> {
+        PlayerCardController controller = uiService.loadFxml("theme/player_card.fxml");
+        controller.setPlayer(player.getPlayer());
+        controller.setFaction(player.getFaction());
+        controller.avatarStackPane.setVisible(false);
+        controller.avatarStackPane.setManaged(false);
+        controller.avatarImageView.setManaged(false);
```

Comment:
The UI elements are only public to allow fxml to set them. If we want an external element to be able to control their state we should introduce a setter.